File: basic/sorting/selection-sort/04-choose-sort.rb
def selection_sort(arr)
  n = arr.length
  (0...n - 1).each do |i|
    min_index = i
    (i + 1...n).each do |j|
      if arr[j] < arr[min_index]
        min_index = j
      end
    end
    # 交换
    arr[i], arr[min_index] = arr[min_index], arr[i]
  end
  arr
end

# 示例
arr = [64, 25, 12, 22, 11]
sorted_arr = selection_sort(arr)
puts "排序后的数组: #{sorted_arr}"

File: basic/sorting/selection-sort/04-choose-sort.js
function selectionSort(arr) {
  let n = arr.length;
  for (let i = 0; i < n - 1; i++) {
      let minIndex = i;
      for (let j = i + 1; j < n; j++) {
          if (arr[j] < arr[minIndex]) {
              minIndex = j;
          }
      }
      // 交换
      [arr[i], arr[minIndex]] = [arr[minIndex], arr[i]];
  }
  return arr;
}

// 示例
let arr = [64, 25, 12, 22, 11];
let sortedArr = selectionSort(arr);
console.log("排序后的数组:", sortedArr);
